### PR TITLE
Add R9SLIM+ target

### PR DESCRIPTION
**Pull request by cruwaller**
R9SLIM+ receiver can be used now. 
This enabled dual bootloader support and build target.

Special thanks to @Rogero- for tracing pins and initial configuration!
This is also verified to work by @Rogero-!

### CODE REVIEW

File: src/src/rx_main.cpp [hunk 1]
```diff
@@ -568,30 +576,43 @@ void ICACHE_RAM_ATTR TXdoneISR()
...
 #ifdef PLATFORM_STM32
+#if defined(TARGET_R9SLIMPLUS_RX)
+    CRSF_RX_SERIAL.setRx(GPIO_PIN_RCSIGNAL_RX);
+    CRSF_RX_SERIAL.begin(CRSF_RX_BAUDRATE);
+
+    Serial.setTx(GPIO_PIN_RCSIGNAL_TX);
+#else /* !TARGET_R9SLIMPLUS_RX */
     #ifdef USE_R9MM_R9MINI_SBUS
-    HardwareSerial(USART2);
+    //HardwareSerial(USART2); // This is useless call
```
Comment (JyeSmith) [inline src/src/rx_main.cpp:588]: Have you checked this is not needed and working on the r9mm?  When it was added, for what ever reason it was needed along with the setTx/Rx calls.

File: src/src/targets.h [hunk 2]
```diff
@@ -95,32 +97,47 @@
...
     #define GPIO_PIN_BUTTON         PB0  // pullup e.g. LOW when pressed
-#else
+#elif TARGET_R9SLIMPLUS_RX
+    #define GPIO_PIN_LED            PA11 // Red
+    #define GPIO_PIN_LED_RED        PA11 // Red
+    #define GPIO_PIN_LED_GREEN      PA12 // Green
+    #define GPIO_PIN_BUTTON         PC13  // pullup e.g. LOW when pressed
+    /* PB3: RX = HIGH, TX = LOW */
+    #define GPIO_PIN_RX_ENABLE PB3
+    /* PB9: antenna 1 (left) = HIGH, antenna 2 (right) = LOW
```
Comment (JyeSmith) [inline src/src/targets.h:135]: This should not be left floating and be set to a single antenna.

File: src/src/rx_main.cpp (hunk 1 above)
Comment (cruwaller) [reply src/src/rx_main.cpp:588]: No need to even check :)
That is the C++ contructor call and does not effect any of the available `Serials`. 
Correct USART peripheral is selected by using defined TX and RX in `Begin` method according to variant `PeripheralPins.c` definitions.
Fortunately linked ignore it as a unused variable...

File: src/src/targets.h (hunk 2 above)
Comment (cruwaller) [reply src/src/targets.h:135]: Ok :)

Comment (cruwaller) [reply src/src/targets.h:135]: fixed

**Comment by AlessandroAU** [comment]
I will check it doesn't break anything and then merge

**Comment by Rogero-** [comment]
> I will check it doesn't break anything and then merge

please wait till we do more antenna selection testing tonight to confirm how it works 100%

**Comment by AlessandroAU** [comment]
Don't worry I talked to PeHo :) It will be a few days until I can test

**Comment by Rogero-** [comment]
> Don't worry I talked to PeHo :) It will be a few days until I can test

many tests were concluded tonight, until the ipex connectors got loose :D 
so the code can actually select which is the active antenna (toggled LOW/HIGH and active antenna was switched),
but with or without setting an active antenna manually in the code, the RSSI output value is always the same (around -40).
The Right Antenna is selected as active by default now, users can use only 1 antenna on the Slim+ 
and the PR is ready to merge :) 
Thank you all for this amazing project.

**Comment by dimitroffs** [comment]
Successfully tested on my r9slim+ - flashed by dual boot loader, binds as expected and sticks checked in bf. Works as expected! Great work guys! More tests to come! We could also think of more interesting stuff using diversity on Rx side ;) Happy to be part of all that!

**Comment by AlessandroAU** [comment]
Ready for merge? 👍

**Comment by cruwaller** [comment]
Yes, good to go :)

**Comment by stkenneth** [comment]
im sorry if i should not ask here, i am new to github...
how to switch the antenna from right to left? which file should be modded? can I pm someone so if this might not be the right place to ask such question?

**Comment by JyeSmith** [comment]
@stkenneth here as an issue or on discord is best.

Change the below from LOW to HIGH will change to the other antenna.

`digitalWrite(GPIO_PIN_ANTENNA_SELECT, LOW);`

https://github.com/AlessandroAU/ExpressLRS/blob/bef20aca72cb92718971413ae0f51f3022a097be/src/lib/SX127xDriver/SX127xHal.cpp#L52

**Comment by stkenneth** [comment]
@JyeSmith thankyou very much! sorry for messing yhe issue tabs...

**Comment by nik012003** [comment]
Does this firmware work also on R9 Slim+ OTA?

**Comment by stkenneth** [comment]
@nik012003 have you try it? i tried with 2 slim+ota and unable to flash through betaflight...

**Comment by Rogero-** [comment]
> @nik012003 have you try it? i tried with 2 slim+ota and unable to flash through betaflight...

you mean the Bootloader flashed and works fine but vscode can't upload the ELRS firmware ? you can have more help with this on Discord :)

**Comment by NiklasVoigt** [comment]
its important to have this things set:
set serialrx_inverted = OFF
set serialrx_halfduplex = OFF

**Comment by stkenneth** [comment]
> its important to have this things set:
set serialrx_inverted = OFF
set serialrx_halfduplex = OFF
@Rogero- yes @PeHo was helping me but we have not found solution yet
@NiklasVoigt when serialrx inverted is on, and then vscode attempt to put it in bootloader mode, and i plug battery, the greenlight on the rx light up and stay on. but vscode keep attempting and fail. while when i turned off serialrx_inverted, and vscode attempt to put bootloader mode and i plug battery in, red light blink for a second and stayed off, also vscode keep attempt and fail.

**Pull request merged**